src/tools/ScenarioMeasurement/Startup/GenericStartupParser.cs

```diff
@@ -0,0 +1,104 @@
+﻿using Microsoft.Diagnostics.Tracing;
+using Microsoft.Diagnostics.Tracing.Parsers;
+using Microsoft.Diagnostics.Tracing.Session;
+using Reporting;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace ScenarioMeasurement
+{
+    // [EventSource(Guid = "9bb228bd-1033-5cf0-1a56-c2dbbe0ebc86")]
+    // class PerfLabGenericEventSource : EventSource
+    // {
+    //     public static PerfLabGenericEventSource Log = new PerfLabGenericEventSource();
+    //     public void Startup() => WriteEvent(1);
+    // }
+
+    internal class GenericStartupParser : IParser
+    {
+        public void EnableKernelProvider(TraceEventSession kernel)
+        {
+            kernel.EnableKernelProvider((KernelTraceEventParser.Keywords)(KernelTraceEventParser.Keywords.Process | KernelTraceEventParser.Keywords.Thread | KernelTraceEventParser.Keywords.ContextSwitch));
+        }
+
+        public void EnableUserProviders(TraceEventSession user)
+        {
+            user.EnableProvider("PerfLabGenericEventSource");
+        }
+
+        public IList<Counter> Parse(string mergeTraceFile, string processName)
+        {
+            var results = new List<double>();
+            var threadTimes = new List<double>();
+            double threadTime = 0;
+            var ins = new Dictionary<int, double?>();
+            double start = -1;
+            int? pid = null;
+            using (var source = new ETWTraceEventSource(mergeTraceFile))
+            {
+
+                source.Kernel.ProcessStart += evt =>
+                {
+                    if (evt.ProcessName.Equals(processName, StringComparison.OrdinalIgnoreCase))
+                    {
+                        if (pid.HasValue)
+                        {
+                            throw new Exception("found a start when we already had a start");
```

Comment:
Do we want to handle cases where there are multiple processes spawned but we care about a specific one?  For example, the SDK?

Reply:
I've thought about this a lot. There's two common buckets of this:

1) A process starts many processes to finish a job (dotnet, roslyn, VS). In those cases the customer experience we're trying to measure is the outermost process. That's what this code currently does. If we wanted to log sub-process splits and have it show up in the reporting tools, we could build that feature. My take is that you'll see that easily enough in the profile, but I could be convinced pretty easily.
2) App models which have other launchers (`dotnet run`, UWP apps?) My intent right now is that we will publish all our tests so we have the minimal host experience, and not use `dotnet run`. (I recognize the importance of `dotnet run` and think we should test it in the context of our SDK testing.) When we start to support UWP apps, depending on the hosting model, we may need to implement the ability to specify "host process" versus "target process" so we report the correct one. We built that into the last version of these tools.

In both cases I think this is complexity we can add as needed.

Reply:
I'm with you that scenario 2 is reasonable to bring up if/when we need it.  For scenario 1, unless I've missed something I don't think it works today.  If you were to run `dotnet build` it would look for a matching process name before process ID and if there were multiple dotnet processes then it would throw.  This would not be an issue for other scenarios that don't spawn processes of their own.

Reply:
Ah, good catch. I'll fix it.